lib/AST/SubstitutionMap.cpp

```diff
@@ -395,6 +398,40 @@ SubstitutionMap::combineSubstitutionMaps(const SubstitutionMap &firstSubMap,
     });
 }
 
+void SubstitutionMap::verify() const {
+#ifndef NDEBUG
```

Comment:
Maybe the entire method should be wrapped in the #ifndef? What do you think @jckarter